Kudu.Core/Jobs/ContinuousJobRunner.cs

```diff
@@ -297,5 +302,21 @@ protected virtual void Dispose(bool disposing)
                 }
             }
         }
+
+        private int GetAvailablePortForWebJob()
+        {
+            if (JobSettings.GetIsWebHook())
+            {
+                TcpListener listener = new TcpListener(IPAddress.Loopback, 0);
+                listener.Start();
+                int port = ((IPEndPoint)listener.LocalEndpoint).Port;
+                listener.Stop();
```

Comment:
Maybe stop in a finally, to be really sure we don't leak anything? Plus this way you don't need the `port` variable at all...